src/util/lang_class.js

```diff
@@ -2,6 +2,23 @@
 
   var slice = Array.prototype.slice, emptyFunction = function() { },
 
+      makePropertyDescriptor = function(value) {
+        return {
+          value: value,
+          writable: true,
+          enumerable: true,
+          configurable: true
+        };
+      },
+
+      getOwnPropertyDescriptor = (function() {
+        function getOwnPropertyDescriptorPolyfill(obj, prop) {
+          return (prop in obj) ? makePropertyDescriptor(obj[prop]) : undefined;
+        }
+
+        return Object.getOwnPropertyDescriptor || getOwnPropertyDescriptorPolyfill;
+      })(),
```

Comment:
The `getOwnPropertyDescriptor` is partially supported in IE8, although I could not find information regarding what is actually implemented and what is not implemented. Thankfully, IE9 fully supports this method.

For the sake of supporting older browsers, we can drop some functionality in favour of compatibility. Any property will be treated as if it were declared as follows:
```javascript
var source = {};
source.foo = 'bar';
```
This inevitably means that the descriptor adds no direct benefit if polyfilled, but we're still using it to advocate consistency.